tests/virt_autotest/sriov_network_card_pci_passthrough.pm

```diff
@@ -215,48 +218,63 @@ sub prepare_guest {
     my $vm = shift;
 
     assert_script_run "virsh dumpxml $vm > $vm.xml";
-    script_run "virsh destroy $vm";
-    assert_script_run "virsh undefine $vm";
-
-    #extra process for XEN hypervisor
-    if (is_xen_host) {
-
-        #enable pci-passthrough and set model for pv guest.
-        #refer to bug #1167217 for the reason
-        my $passthru_xml = "<passthrough state='on'/>";
-        my $e820_xml     = "";
-        if (is_pv_guest($vm)) {
-            $e820_xml = "<e820_host state='on'/>";
-        }
-        if (script_run("grep '<features>' $vm.xml") == 0) {
-            assert_script_run "sed -i \"/<features>/a\\<xen>\\n$passthru_xml\\n$e820_xml\\n</xen>\" $vm.xml";
-        }
-        else {
-            assert_script_run "sed -i \"/<domain /a\\<features>\\n<xen>\\n$passthru_xml\\n$e820_xml\\n</xen>\\n</features>\" $vm.xml";
-        }
-
-        #disable memory ballooning for fv guest as it is not supported
-        if (is_fv_guest($vm)) {
-            assert_script_run "sed -i '/<currentMemory/d' $vm.xml";
+    my $mem          = "";
+    my $cur_mem      = "";
+    my $is_ballooned = 0;
+    $mem          = script_output "xmlstarlet sel -t -v //memory $vm.xml";
+    $cur_mem      = script_output "xmlstarlet sel -t -v //currentMemory $vm.xml";
+    $is_ballooned = 1 if $cur_mem != $mem;
```

Comment:
If we're leaving it here, why not simply do
```suggestion
    my $is_ballooned = ($cur_mem != $mem);
```
if it will be moved into a lib function, please ignore this suggestion

Reply:
it will be moved into a lib function, but yes it is more concise.